src/modules/web-inbox/helper.js

```diff
@@ -38,6 +39,35 @@ export const addWebInbox = (logger) => {
   document.body.appendChild($ct.inbox)
 }
 
+const getAndMigrateInboxMessages = (guid) => {
+  const messages = StorageManager.readFromLSorCookie(WEBINBOX) || {}
+  // Doing this to migrate message to guid level
+  if (Object.keys(messages).length > 0 && Object.keys(messages)[0].includes('_')) {
+    const gudInboxObj = {}
+    gudInboxObj[guid] = messages
+    StorageManager.saveToLSorCookie(WEBINBOX, gudInboxObj)
+    return gudInboxObj
+  }
+  return messages
+}
+
+export const getInboxMessages = () => {
+  const guid = JSON.parse(decodeURIComponent(StorageManager.read(GCOOKIE_NAME)))
+  if (!isValueValid(guid)) { return {} }
+  const messages = getAndMigrateInboxMessages(guid)
+
+  return messages.hasOwnProperty(guid) ? messages[guid] : {}
+}
+
+export const saveInboxMessages = (messages) => {
+  const guid = JSON.parse(decodeURIComponent(StorageManager.read(GCOOKIE_NAME)))
+  if (!isValueValid(guid)) { return }
+  const storedInboxObj = getAndMigrateInboxMessages(guid)
+
+  const newObj = { ...storedInboxObj, [guid]: messages }
```

Comment:
let's say your 
`storedInboxObj = {
    '1234' : {
        '5_6' : {}
    }
}`

where guid = '1234' and the id of the existing msg in the inbox is '5_6'

Let's assume the new message that you've to add is 
`messages = {
    '7_8': {},
    '9_10': {}
}`

when you do - `newObj = { ...storedInboxObj, [guid]: messages }`

your output would be newObj = `{
    '1234': {
        '7_8': {},
        '9_10': {}
    }
}`

Reply:
you'll end up losing message with id = '5_6'

Reply:
Since the messages is the reference to the cache , which is updated when new message is received . The newObj will get the updated messages

Reply:
Where ever saveInboxMessages is being called, we first read the messages, make changes to the messages object and save the changed object itself.

Reply:
Approved based on the discussion with Sonam that every time the saveInboxMessages function is called we always pass all the messages for a guid and not just the increments / delta.